Library/OcMainLib/OpenCoreKernel.c

```diff
@@ -1136,6 +1146,31 @@ OcKernelFileOpen (
     && Status == EFI_NOT_FOUND
     && OpenMode == EFI_FILE_MODE_READ
     && (StrStr (FileName, L"\\kernelcache") != NULL)) {
+    //
+    // Change the target to the custom one if requested CustomKernel.
+    //
+    if (mCustomKernelDirectory != NULL) {
+      DEBUG ((DEBUG_INFO, "OC: Redirecting %s to the custom one on ESP\n", FileName));
+      NewFileName = OcStrrChr (FileName, L'\\');
+      if (NewFileName == NULL) {
+        NewFileName = FileName;
+      }
+
+      DEBUG ((DEBUG_INFO, "OC: Filename after redirection: %s\n", NewFileName));
+
+      mCustomKernelDirectoryInProgress = TRUE;
```

Comment:
This change is not correct, as currently there is no opened file, and we need to do fuzzy matching in the custom directory via OcKernelFuzzyMatch. As a result no opening should happen here, only directory name swapping.

Reply:
```C
This = mCustomKernelDirectory;
FileName = NewFileName;
```

This is what should be enough.